tests/test_feature_selection.py

```diff
@@ -0,0 +1,101 @@
+# Authors: Soledad Galli <solegalli1@gmail.com>
+# License: BSD 3 clause
+import pytest
+import pandas as pd
+
+from feature_engine.feature_selection import DropFeatures
+
+
+def test_FeatureEliminator_initial(dataframe_vartypes, dataframe_na):
+    # test case 1: with dataframe_vartypes
+    transformer = DropFeatures(features_to_drop=["City", "dob"])
+    X = transformer.fit_transform(dataframe_vartypes)
+
+    # expected result
+    df = pd.DataFrame(
+        {'Name': ['tom', 'nick', 'krish', 'jack'],
+         'Age': [20, 21, 19, 18],
+         'Marks': [0.9, 0.8, 0.7, 0.6]}
+    )
+
+    # init params
+    assert transformer.features == ['City', 'dob']
+    # transform params
+    assert X.shape == (4, 3)
+    assert type(X) == pd.DataFrame
+    pd.testing.assert_frame_equal(X, df)
+
+
+def test_FeatureEliminator_non_existing_variables(dataframe_vartypes):
+    # test case 2: passing variables that doesn't exist
+    with pytest.raises(KeyError):
+        transformer = DropFeatures(features_to_drop="last_name")
+        X = transformer.fit_transform(dataframe_vartypes)
+
+
+def test_FeatureEliminator_non_existing_index(dataframe_vartypes):
+    # test case 3: passing an index inside the list
+    with pytest.raises(KeyError):
+        transformer = DropFeatures(features_to_drop=[10])
+        X = transformer.fit_transform(dataframe_vartypes)
+
+
+def test_FeatureEliminator_integer_as_variable(dataframe_vartypes):
+    # test case 4: with integer as feature_to_drop
+    with pytest.raises(KeyError):
+        transformer = DropFeatures(features_to_drop=99)
+        X = transformer.fit_transform(dataframe_vartypes)
+
+
+def test_FeatureEliminator_different_input():
```

Comment:
here you are testing an error when input is not a dataframe?

can we rename the test to test_drop_features_errors_if_fit_input_not_dataframe? or if transform_input_not_dataframe, and maybe separate fit and transform? it should error in both